infra/proposer.go

```diff
@@ -45,22 +51,25 @@ func CreateProposer(addr string, crypto *Crypto) *Proposer {
 		panic(err)
 	}
 
-	return &Proposer{e: endorser}
+	return &Proposer{e: endorser, addr: addr}
 }
 
-func (p *Proposer) Start(signed, processed chan *Elecments, done <-chan struct{}) {
+func (p *Proposer) Start(signed, processed chan *Elecments, done <-chan struct{}, size int) {
 	for {
 		select {
 		case s := <-signed:
+			//fmt.Println("send to " + p.addr)
 			r, err := p.e.ProcessProposal(context.Background(), s.SignedProp)
 			if err != nil || r.Response.Status < 200 || r.Response.Status >= 400 {
-				fmt.Printf("Err processing proposal: %s, status: %d\n", err, r.Response.Status)
+				fmt.Printf("Err processing proposal: %s, status: %d, addr: %s \n", err, r.Response.Status, p.addr)
+				fmt.Println(r)
 				continue
 			}
-
-			s.Response = r
-			processed <- s
-
+			s.Response = append(s.Response, r)
+			if len(s.Response) == size {
+				//fmt.Println("Sending for block")
+				processed <- s
+			}
```

Comment:
this is not thread-safe: multiple go routines `append` and `len` the `s.Response` at the same time.

Reply:
Use channel instead.